testapps/test_pillowtop/tests/test_case_pillow.py

```diff
@@ -82,21 +87,47 @@ def test_case_soft_deletion(self):
         # verify there
         results = CaseES().run()
         self.assertEqual(1, results.total)
+        search_results = CaseSearchES().run()
+        self.assertEqual(1, search_results.total)
 
         # soft delete the case
         with self.process_case_changes:
             CommCareCase.objects.soft_delete_cases(self.domain, [case_id])
         self.elasticsearch.indices.refresh(CASE_INDEX_INFO.index)
+        self.elasticsearch.indices.refresh(CASE_SEARCH_INDEX_INFO.index)
+
+        # ensure not there anymore
+        results = CaseES().run()
+        self.assertEqual(0, results.total)
+        search_results = CaseSearchES().run()
+        self.assertEqual(0, search_results.total)
+
+    def test_case_hard_deletion(self):
+        case_id, case_name = self._create_case_and_sync_to_es()
+
+        # verify there
+        results = CaseES().run()
+        self.assertEqual(1, results.total)
+        search_results = CaseSearchES().run()
+        self.assertEqual(1, search_results.total)
+
+        # hard delete the case
+        with self.process_case_changes:
+            CommCareCase.objects.hard_delete_cases(self.domain, [case_id])
+        self.elasticsearch.indices.refresh(CASE_INDEX_INFO.index)
+        self.elasticsearch.indices.refresh(CASE_SEARCH_INDEX_INFO.index)
 
         # ensure not there anymore
         results = CaseES().run()
         self.assertEqual(0, results.total)
+        search_results = CaseSearchES().run()
+        self.assertEqual(0, search_results.total)
 
     def _create_case_and_sync_to_es(self):
         case_id = uuid.uuid4().hex
         case_name = 'case-name-{}'.format(uuid.uuid4().hex)
         with self.process_case_changes:
-            create_and_save_a_case(self.domain, case_id, case_name)
+            create_case(self.domain, case_id=case_id, name=case_name, save=True, enable_kafka=True)
```

Comment:
why was it necessary to change the way the cases are created? The previous way was already publishing to kafka.

Reply:
Ah I was following the note in the docstring for `create_and_save_a_case` which said 
> Use corehq.form_processor.tests.utils.create_case() instead if possible

And was determined to switch to that method instead of reverting to the old method when making the additional changes to allow publishing to kafka. The logic with that docstring is that the new method creates cases and forms directly, rather than submitting a form which takes longer so that sounded good to me.